modules/core/keeper/msg_server_test.go

```diff
@@ -2016,3 +2016,77 @@ func (suite *KeeperTestSuite) TestUpdateConnectionParams() {
 		})
 	}
 }
+
+func (suite *KeeperTestSuite) TestPruneStalePacketState() {
+	var msg *channeltypes.MsgPruneStalePacketState
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expErr   error
+	}{
+		{
+			"success",
+			func() {},
+			nil,
+		},
+		{
+			"failure: core keeper function fails, pruning sequence end not found",
+			func() {
+				msg.PortId = "portidone"
+			},
+			channeltypes.ErrPruningSequenceEndNotFound,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			path := ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.Setup(path)
+
+			// configure the channel upgrade version on testing endpoints
+			path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = ibcmock.UpgradeVersion
+			path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = ibcmock.UpgradeVersion
+
+			err := path.EndpointA.ChanUpgradeInit()
+			suite.Require().NoError(err)
+
+			err = path.EndpointB.ChanUpgradeTry()
+			suite.Require().NoError(err)
+
+			err = path.EndpointA.ChanUpgradeAck()
+			suite.Require().NoError(err)
+
+			err = path.EndpointB.ChanUpgradeConfirm()
+			suite.Require().NoError(err)
+
+			err = path.EndpointA.ChanUpgradeOpen()
+			suite.Require().NoError(err)
+
+			err = path.EndpointA.UpdateClient()
+			suite.Require().NoError(err)
+
+			msg = channeltypes.NewMsgPruneStalePacketState(
+				path.EndpointA.ChannelConfig.PortID,
+				path.EndpointA.ChannelID,
+				10,
```

Comment:
should we add a test case that modifies the limit?

EDIT: nvm we test limit in `TestPruneStalePacketState`, I think I'm happy enough with that!